### DISCUSSION TITLE
Pausing functions while Popup Page showing

### DISCUSSION THREAD

**Discussion by Bwanna** [ImpulseAdventure/GUIslice#490] [Q&A]
Hi,
Great Library!

I would like to stop certain programming from running in the void loop() whenever the main display page is not showing (not the 'focus'), e.g. whenever a Popup or Jump page is being shown.
I'm considering the approach of changing the PauseProcess variable (below) whenever the base page is not the focus and another page is being shown.

**Question:**  Any suggestions on the most efficient approach?  e.g. a single place to insert Flags?

**My first idea** is to use the below routine in the main loop and insert "PauseProcess =1" and "PauseProcess =0" flags to affect the below routine.  Rather than placing these flags multiple times throughout the program, is there a single place they could be placed to take effect anytime this occurs - the base page is no longer the focus?

```
void loop() {
  if (PauseProcess = 0) {   //0=not paused,1=paused
     //Code doing things when base page is being shown
   } else if (PauseProcess = 1) {
     //Code doing things when base page not being shown, i.e. ANY popup/jump page being shown
   }
```

**Another idea:**  is there a way to determine the "Modal" status?  Could use this as Flag indicator instead of the PauseProcess variable?  (ref: GUIslice.c)

**Comment by Pconti31** [comment]
@Bwanna I would just check the current page id and go from there
```
void loop() {
  if (gslc_GetPageCur(&m_gui) == E_PG_MAIN) { 
     //Code doing things when base page is being shown
   } else {
     //Code doing things when base page not being shown, i.e. ANY popup/jump page being shown
   }
```
By the way you need to avoid the beginners mistake of using '=' instead of '=='. Since '=' is an assignment.
Paul--

**Reply by Bwanna** [reply]
Thank you!  Very much appreciate the feedback!
